src/state/opamSwitchState.ml

```diff
@@ -304,9 +323,10 @@ let load lock_kind gt rt switch =
           let nv = OpamPackage.create nv.name version in
           let o = OpamFile.OPAM.with_version version o in
           OpamPackage.Set.add nv pinned,
-          OpamPackage.Map.add nv o opams
+          OpamPackage.Map.add nv o opams,
+          (OpamFile.OPAM.depexts o) <> [] || depexpts_present
```

Comment:
Here, you only check that the field is not empty. The field can be non empty, but resolved to the current setup the list is empty. It worth using `depexts`